otter/log/spec.py

```diff
@@ -0,0 +1,118 @@
+"""
+Format logs with specification
+"""
+
+from copy import deepcopy
+
+from otter.constants import ServiceType
+from otter.log.formatters import IGNORE_FIELDS
+
+
+class UUID(basestring):
+    """
+    UUID
+    """
+
+
+class IPAddress(basestring):
+    """
+    IP Address
+    """
+
+
+# Allowed fields with their types
+fields = {
+    "num_servers": int,
+    "server_id": UUID,
+    "clb_id": int,
+    "ip_address": IPAddress,
+}
+
+
+# Fields that will occur in error
+error_fields = {
+    "system": SystemType,
+    "operation": basestring,
+    "url": basestring,
+    "code": int,
+    "message": basestring,
+    "body": basestring,
+    "headers": dict     # need more validation. I miss type system :(
+}
+
+
+# mapping from msg type -> message
+msg_types = {
+    "launch-servers": "Launching {num_servers} servers",
+    "delete-server": "Deleting {server_id} server",
+    "add-server-clb": ("Adding {server_id} with IP address {ip_address} "
+                       "to CLB {clb_id}"),
+    "remove-server-clb": ("Removing server {server_id} with IP address "
+                          "{ip_address} from CLB {clb_id}"),
+}
+
+
+def SpecificationObserverWrapper(observer):
+    """
+    Return observer that validates messages based on specification
+    and delegates to given observer
+    """
+    def validate_observer(event_dict):
+        try:
+            speced_event = validate_message(event_dict)
+        except ValueError:
+            # TODO: What to do if it is not valid? Should it instead
```

Comment:
That seems like a workable solution - we can look for these in logs and slowly work to fix them.  :)